ouroboros-consensus/src/unstable-consensus-testlib/Test/CsjModel/NonEmptySeq.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE BangPatterns #-}
+
+-- | A non-empty sequence type with some custom methods with types that are
+-- convenient for "Test.CsjModel"
+module Test.CsjModel.NonEmptySeq (
+    module Test.CsjModel.NonEmptySeq
+  ) where
+
+import           Control.Monad (guard)
+import qualified Data.Maybe as L (Maybe (Just, Nothing))
+import           Data.Sequence (Seq)
+import qualified Data.Sequence as Seq
+
+-- | A non-empty 'Seq'
+newtype NonEmptySeq a = UnsafeNonEmptySeq (Seq a)
```

Comment:
This type doesn't use `AnchoredFragment` for two reasons.

- As a model, I'm trying to keep it as simple as possible: so far, I don't need headers at all. And I can't put points in an `AnchoredFragment` because of the `HasHeader` constraint.

- So far, the model deals with chains that reach all the way back to `Origin`, not _fragments_. As long as that's true, it seems misleading to use `AnchoredFragment`.

Reply:
The code no longer reaches all the way back to Genesis. It still doesn't involve headers explicitly, but perhaps `AnchoredSeq` could still be reused?